src/blockchain/electrum.rs

```diff
@@ -385,4 +386,116 @@ mod test {
 
         assert_eq!(wallet.get_balance().unwrap(), 50_000);
     }
+
+    #[test]
+    fn test_electrum_blockchain_factory_sync_with_stop_gaps() {
```

Comment:
can we put this in `blockchain_tests` so it covers all blockchains?

Reply:
That's a good idea, I'll do so in a bit.

Reply:
> can we put this in `blockchain_tests` so it covers all blockchains?

I feel like this is too much work for this PR. Should we get this merged and I can create a second ticket?

Each test case requires creating a new `Blockchain` with a specified `stop_gap`.

I am also not very familiar with macros in Rust, so this may take a while.

Reply:
Hmmmm. Ok I see and agree with this point. Not all blockchains have "stop gaps" anyways.

Reply:
We can definitely do something universal I think. Can create a custom test trait with a method such as `generate_blockchain_with_stop_gap` and have that trait as an input.

Reply:
I've done something relatively universal, we can continue this conversation here: https://github.com/bitcoindevkit/bdk/pull/652#issuecomment-1173789385